Fix condition for streaming output_ids in tokenizer manager

merrymercy opened the pull request:
There is a bug in how we handle the `output_ids` for streaming requests. (python/sglang/srt/managers/tokenizer_manager.py)

In the default case, for each chunk, the `output_ids` field should contain all output token ids, regardless of streaming or not
Only when users specify `--stream-output`, for each chunk, the `output_ids` field will contain delta output tokens ids.

This logic is correctly handled in the branch of `elif isinstance(recv_obj, BatchTokenIDOutput):`

However, in the branch `isinstance(recv_obj, BatchStrOutput):`, the code was wrong in the very beginning when it was introduced (https://github.com/sgl-project/sglang/pull/8837/files#r2552360555).
It later leads to some very hacky PRs like https://github.com/sgl-project/sglang/pull/9467. (not sure if this is the root cause).

This might be a breaking change but it should be a correct fix. I need more help from @JustinTong0323 @CatherineSue to see whether it fixes or breaks any gpt oss things. 


### Other minor fix
- fix the print in python/sglang/bench_one_batch_server.py
- cache the check of `is_gpt_oss`

Thanks @JeffryLee for finding this bug.

merrymercy committed:
Simplify the check on is_gpt_oss

merrymercy committed:
Fix condition for streaming output in tokenizer manager

merrymercy committed:
Fix bench one batch server

Commit:
Remove hacky check for 20006 and update ResponseOutputText call to match with latest openai

CatherineSue reviewed (approved):
Tested with `python3 -m sglang.launch_server --model /models/openai/gpt-oss-120b --port=8034 --tp-size=4 --reasoning-parser=gpt-oss --stream-output`

Hardware: H100

It works with the changes in b793a83: 1) hacky changes removed. 2) Update `ResponseOutputText` to use `text` instead of `content` field as it has been removed.

streaming:
<img width="1295" height="635" alt="Image" src="https://github.com/user-attachments/assets/f6467272-1ff8-4c9c-a47e-aacdb60804aa" />

**Note:** streaming won't work without `--stream-output`

non-streaming:
<img width="1299" height="250" alt="Screenshot 2025-11-24 at 5 39 10 PM" src="https://github.com/user-attachments/assets/467367bf-bbc1-414b-9a2e-ed840c71489d" />

openai 2.6.1 API:
<img width="616" height="244" alt="Image" src="https://github.com/user-attachments/assets/5d6fdc5e-d8ba-4499-aab7-1ff3d9a3115a" />

JustinTong0323 committed:
Merge branch 'main' into lianmin/fix-stream

JustinTong0323 committed:
fix: enhance StreamingHarmonyContext to handle cumulative tokens processing

Signed-off-by: Xinyuan Tong <xinyuantong.cs@gmail.com>

JustinTong0323 committed:
feat: add streaming response check in BaseTestGptOss and clarify StreamingHarmonyContext token processing

- Implemented a new method to verify streaming responses from the API.
- Updated StreamingHarmonyContext to clarify token processing logic based on streaming output settings.

Signed-off-by: Xinyuan Tong <xinyuantong.cs@gmail.com>
